bin/rule-generator.js

```diff
@@ -0,0 +1,448 @@
+};
+
+const showBanner = () => {
+	const COLOR_DQ_HEX = '#0077c8';
+	console.log(chalk.hex(COLOR_DQ_HEX)(figlet.textSync('Axe Rule Generator')));
+};
+
+const QUESTIONS = {
+	GET_RULE_NAME: {
+		/**
+		 * What is the name of the RULE? (Eg: aria-valid):
+		 * 	- check if there is conflict in rule (file name)
+		 * 	- also if there is conflict in rule id itself
+		 */
+		name: 'GET_RULE_NAME',
+		type: 'input',
+		message: 'What is the name of the RULE? (Eg: aria-valid):',
+		validate: input => {
+			const ruleName = input.toLowerCase();
+			// 1) check if valid rule name
+			if (
+				!(
+					ruleName &&
+					ruleName.length > 0 &&
+					ruleName.search(/^[a-zA-Z-]+$/) !== -1
+				)
+			) {
+				throw new Error(
+					'RULE name should not be empty and can contain alphabets and dashes.'
+				);
+			}
+			// 2) ensure no rule filename overlaps
+			if (axeRulesFileNames.includes(ruleName)) {
+				throw new Error('RULE name conflicts with an existing filename.');
+			}
+			// 3) ensure no rule id overlaps
+			if (axeRulesIds.includes(ruleName)) {
+				throw new Error('Rule ID already exists.');
+			}
+			return true;
+		}
+	},
+	GET_IS_RULE_MATCHES: {
+		/**
+		 * Does the rule need a matches file to be created? (Yes/ No):
+		 */
+		name: 'GET_IS_RULE_MATCHES',
+		type: 'confirm',
+		message: 'Does the RULE need a MATCHES file to be created?:'
+	},
+	GET_IS_CHECK: {
+		/**
+		 *  Would you like to create a CHECK for the rule? (Yes/ No):
+		 */
+		name: 'GET_IS_CHECK',
+		type: 'confirm',
+		message: 'Would you like to create a CHECK for the RULE?:'
+	},
+	GET_IS_ANOTHER_CHECK: {
+		/**
+		 *  Would you like to create a CHECK for the rule? (Yes/ No):
+		 */
+		name: 'GET_IS_ANOTHER_CHECK',
+		type: 'confirm',
+		message: 'Would you like to create another CHECK for the RULE?:'
+	},
+	GET_CHECK_NAME: {
+		/**
+		 * Enter name of CHECK for the RULE?
+		 * 	(Eg: aria-label)
+		 * 	- check if there is conflict in rule (file name)
+		 * 	- also if there is conflict in rule id itself
+		 */
+		name: 'GET_CHECK_NAME',
+		type: 'input',
+		message: 'Enter name of CHECK for the RULE? (Eg: aria-label): ',
+		validate: input => {
+			const checkName = input.toLowerCase();
+			// 1) check if valid rule name
+			if (
+				!(
+					checkName &&
+					checkName.length > 0 &&
+					checkName.search(/^[a-zA-Z-]+$/) !== -1
+				)
+			) {
+				throw new Error(
+					'CHECK name should not be empty and can contain alphabets and dashes.'
+				);
+			}
+			// 2) ensure no rule filename overlaps
+			if (axeChecksFileNames.includes(checkName)) {
+				throw new Error('CHECK name conflicts with an existing filename.');
+			}
+			// 3) ensure no rule id overlaps
+			if (axeChecksIds.includes(checkName)) {
+				throw new Error('CHECK ID already exists.');
+			}
+			return true;
+		}
+	},
+	GET_CHECK_CATEGORY: {
+		/**
+		 * Choose category for the CHECK? :
+		 * 	(Eg: aria, color, forms, label, language, media...)
+		 */
+		name: 'GET_CHECK_CATEGORY',
```

Comment:
What are these redundant name properties for?

Reply:
@marcysutton I do not follow, which of these is redundant?

Reply:
The object and the `name` property. I couldn't find where `name` was used.